### DISCUSSION TITLE
Advanced Inference Graph usecase

### DISCUSSION THREAD

**Discussion by MichalPitr** [kubeflow/kfserving#2516] [Q&A]
Hi folks, 

as a part of my research thesis, I'm trying to model parallel paths in an inference graph. In the below example, I have some user features that I'd like to use in 2 ways: 
1. Retrieve related item features
2. Directly pass the user features onto the next part of the inference pipeline. (We can also imagine doing some more parallel processing here)

Since these two paths are asynchronous, I suspect I'll need some way to then synchronize and combine the results from each sub-path. 

Any tips on how to achieve this? I'd appreciate any pointers.
Many thanks

![image](https://user-images.githubusercontent.com/21157924/201094430-2e9388c0-eba0-43cd-b1c7-e2f68b0e004d.png)

**Comment by jagadeeshi2i** [comment]
@MichalPitr Did you try something like this - https://pastebin.com/raw/hnSXXrnH 

https://github.com/kserve/kserve/tree/master/docs/samples/graph.
